src/Batch/Batch/Certificates/NewBatchCertificateCommand.cs

```diff
@@ -38,15 +38,26 @@ public class NewBatchCertificateCommand : BatchObjectModelCmdletBase
         public byte[] RawData { get; set; }
 
         [Parameter]
-        [ValidateNotNullOrEmpty]
+        [ValidateNotNull]
         public SecureString Password { get; set; }
 
+        [Parameter]
+        [ValidateNotNull]
+        public PSCertificateKind? Kind { get; set; }
```

Comment:
There's no need to use nullable types for optional parameters (a `[Parameter]` is by default not mandatory)

Reply:
The issue here is that I really need 3 states.

1. `Kind == null` which means "use the old defaulting behavior, CER if no password, PFX if password"
2. `Kind == CER` - the data is a certificate.
3. `Kind == PFX` - The data is a PFX.

If I make this non-nullable I need some other way to determine if this parameter was specified or not - it seemed easier to just let the parameter itself support 3 states, "not specified" (null), and the other two.

If there's a better way to do this let me know (see below in the `ExecuteCmdletImpl` where this logic is applied and you see the defaulting behavior).

Reply:
I think the problem with a **nullable + optional** parameter is the user may think that passing `null` or not specifying this param may have different results.

Besides, making it not-nullable, i.e. changing the first state to "user not specify Kind" won't make the code much more complex:
```diff
- this.Kind ?? (password == null ? PSCertificateKind.Cer : PSCertificateKind.Pfx),
+ this.IsParameterBound(c => c.Kind) ? (password == null ? PSCertificateKind.Cer : PSCertificateKind.Pfx) : null
```

Reply:
Thanks @isra-fel - I was not familiar with `IsparameterBound` - will use that isntead.

Reply:
I pushed an update which does as you suggested @isra-fel